seacrowd/sea_datasets/id_msvd/id_msvd.py

```diff
@@ -0,0 +1,138 @@
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = TASK_TO_SCHEMA[_SUPPORTED_TASKS[0]].lower()  # "vidtext"
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "video_path": datasets.Value("string"),
+                    "texts": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = SCHEMA_TO_FEATURES[self.SEACROWD_SCHEMA_NAME.upper()]  # video_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        # expect several minutes to download video data ~1.7GB
+        data_path = dl_manager.download_and_extract(_URLS)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "text_path": Path(data_path["text"]),
+                    "video_path": Path(data_path["video"]) / "YouTubeClips",
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, text_path: Path, video_path: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        text_data = []
+        with open(text_path, "r", encoding="utf-8") as f:
+            for line in f:
+                id = line.find(" ")
+                video = line[:id]
+                text = line[id+1:].strip()
+                text_data.append([video, text])
+
+        df = pd.DataFrame(text_data, columns=["video_path", "text"])
+        df["video_path"] = df["video_path"].apply(lambda x: video_path / f"{x}.avi")
+
+        if self.config.schema == "source":
+            df = df.groupby("video_path").agg(list).reset_index()
+            for i, row in df.iterrows():
+                yield i, {
+                    "video_path": str(row["video_path"]),
+                    "texts": row["text"],
```

Comment:
hence, in SEACrowd Source Implementation, we shouldn't group by its video path, because it will alter it from the source data
```suggestion
            for i, row in df.iterrows():
                yield i, {
                    "video_path": str(row["video_path"]),
                    "text": row["text"],
```